source/module_hamilt_lcao/module_deepks/LCAO_deepks_interface.cpp

```diff
@@ -79,7 +80,56 @@ void LCAO_Deepks_Interface::out_deepks_labels(double etot,
             {
                 ld->save_npy_o(deepks_bands, "o_base.npy", nks); // no scf, o_tot=o_base
             }                                                    // end deepks_scf == 0
-        }                                                        // end bandgap label
+        }                                                        // end bandgap label                                                  
+        if(GlobalV::deepks_v_delta)//gamma only now
+        {
+            ModuleBase::matrix h_tot;
+            h_tot.create(nlocal,nlocal);
+
+            ld->collect_h_mat(ld->h_mat,h_tot,nlocal);
+            ld->save_npy_h(h_tot, "h_tot.npy",nlocal);
+
+            if(GlobalV::deepks_scf)
+            {
+                ModuleBase::matrix v_delta;
+                v_delta.create(nlocal,nlocal);
+                ld->collect_h_mat(ld->H_V_delta,v_delta,nlocal);
+                ld->save_npy_h(h_tot-v_delta, "h_base.npy",nlocal);
+                ld->save_npy_h(v_delta, "v_delta.npy",nlocal);
+
+                if(GlobalV::deepks_v_delta==1)//v_delta_precalc storage method 1
+                {
+                    ld->cal_v_delta_precalc(nlocal,
+                            nat,
+                            ucell,
+                            orb,
+                            GridD);
+                
+                    ld->save_npy_v_delta_precalc(nat, 1, nlocal);
+                }
+                else if(GlobalV::deepks_v_delta==2)//v_delta_precalc storage method 2
```

Comment:
do not use GlobalV, pass the deeps_v_delta in from the input parameters of function